check_rebase.py

```diff
@@ -0,0 +1,153 @@
+"""
+from os.path import join, isdir
+import sys
+import signal
+import shutil
+import logging
+from psa_builder import *
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[Rebase-Check] %(asctime)s: %(message)s.",
+    datefmt="%H:%M:%S",
+)
+
+
+def _add_remote_repo(repo, remote, dir):
+    """
+    Add a new remote to a cloned repository
+    :param repo: Name of the git repository
+    :param remote: Dictionary containing dependency details
+    :param dir: Directory to perform operation
+    """
+
+    gitref = dependencies[remote].get(repo)[0]
+    gitbranch = dependencies[remote].get(repo)[1]
+
+    logging.info(
+        "Checking if remote name %s exists for %s repo, if not add it..",
+        remote,
+        repo,
+    )
+
+    cmd = ["git", "-C", join(dir, repo), "remote", "show", remote]
+    ret = run_cmd_and_return(cmd)
+    if ret != 0:
+        # Add remote name
+        cmd = ["git", "-C", join(dir, repo), "remote", "add", remote, gitref]
+        ret = run_cmd_and_return(cmd)
+        if ret != 0:
+            logging.critical(
+                "Failed to add remote name - %s, error: %d", remote, ret
+            )
+            sys.exit(1)
+
+        logging.info(
+            "Remote name %s added successfully for %s repo", remote, repo
+        )
+    else:
+        logging.info("Remote name %s exists for %s repo", remote, repo)
+
+
+def _checkout_rebase_branch(repo, dir, branch_name):
+    """
+    Create a local branch for rebasing
+    :param repo: Name of the git repository
+    :param dir: Directory to perform operation
+    """
+
+    cmd = ["git", "-C", join(dir, repo), "checkout", "-B", branch_name]
+    ret = run_cmd_and_return(cmd)
+    if ret != 0:
+        logging.critical("Failed to checkout %s, error: %d", branch_name, ret)
+        sys.exit(1)
+
+    logging.info("Checkout of %s complete..", branch_name)
+
+
+def _perform_rebase(repo, remote_1, remote_2, dir):
+    """
+    Perform a rebase using 2 remotes
+    :param repo: Name of the git repository
+    :param remote_1: Dictionary containing dependency details
+    :param remote_1: Dictionary containing dependency details
+    :param dir: Directory to perform operation
+    """
+
+    gitbranch_1 = dependencies[remote_1].get(repo)[1] + "-rebase"
+    gitbranch_2 = dependencies[remote_2].get(repo)[1]
+
+    check_and_clone_repo(repo, remote_1, TF_M_BUILD_DIR)
+    _checkout_rebase_branch(repo, TF_M_BUILD_DIR, gitbranch_1)
+
+    cmd = ["git", "-C", repo, "rebase", gitbranch_2]
+    ret = run_cmd_output_realtime(cmd, dir)
+    if ret != 0:
+        logging.critical(
+            "Failed to rebase %s on %s, error: %d",
+            (remote_2 + "/" + gitbranch_2),
+            (remote_1 + "/" + gitbranch_1),
+            ret,
+        )
+        sys.exit(1)
+
+    logging.info(
+        "Rebase of %s on %s successfull..",
+        (remote_2 + "/" + gitbranch_2),
+        (remote_1 + "/" + gitbranch_1),
+    )
+
+
+def _setup_tfm_repo():

```

Comment:
Perhaps change the function name to `setup_and_rebase_tfm_repositories`

Reply:
done